doc/sphinx/practical-tools/utilities.rst

```diff
@@ -7,32 +7,307 @@
+
+Once you've installed Coq, you can easily install user-developed packages
+from the `package list <https://coq.inria.fr/opam/www/>`_ or other opam repositories.
+If you install Coq
+using the `Coq Platform <https://github.com/coq/platform>`_, you have options
+to automatically install some of the most frequently used packages at the
+same time.  While there's currently no guarantee that user-developed packages
+will compile on the current version of Coq, all packages
+that Coq platform installs should compile without difficulty--this is part of
+the Coq platform release process.  When you upgrade to a new version of
+Coq platform, the packages it installed will be updated to as needed to
+remain consistent.
+
+Note that packages marked `released` in the package list web page are more stable
+than those marked `extra-dev`.  To install `extra-dev` packages,
+first add the `coq-extra-dev` opam repository to your local opam installation
+with this command::
+
+  opam repo add coq-extra-dev https://coq.inria.fr/opam/extra-dev
 
-   % ocamlfind ocamlopt -thread -rectypes -linkall -linkpkg \
-                 -package coq.toplevel \
-                 topbin/coqtop_bin.ml -o my_toplevel.native
+While this is the easiest way to install packages, it is not the only way.
+
+You can see the list of installed Coq packages with `opam list|grep coq-`, which
+will show you lines like this containing the package name, its version and short
+description::
+
+   $ opam list | grep coq-
+   coq-bignums               8.15.0          Bignums, the Coq library of arbitrary large numbers
+
+However, you will need to find the :term:`logical name`: used to refer to the package
+in :cmd:`Require` commands.  There are a couple ways to do this:
+
+- If you installed with opam, use :n:`opam show --list-files coq-bignums | head -n1` -
+  the last component of the filename is the logical name (`Bignums`).
+
+- On Linux, :n:`ls $(coqtop -where)/user-contrib` shows the logical names of all
+  installed user-contributed packages.  You should be able to guess which one you
+  need.
+
+- Use the :cmd:`Print LoadPath` command when running Coq, which shows the mapping
+  from :term:`logical path`\s to directories.  Again, you should be able to guess.
+
+The last two methods work even if you didn't install with opam.  Perhaps in the
+future the package name to logical name mapping will be more readily available.
+
+Once you know the logical name of the package, use it to load compiled
+files from the package with the :cmd:`Require` command.
 
 
-For example, to statically link |Ltac|, you can just do:
+Setup for working on your own projects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The working and master copies of source code for your own projects should
+be kept separate from the Coq installation.  In particular, when you upgrade
+to a new version of Coq, any directories you created in the old version won't
+be copied or moved.
```

Comment:
Yes, it would be strange.  Just trying to keep beginners out of trouble.

I made it: "The working and master copies of source code for your own projects should
not be in the directory tree where Coq is installed."